File: ch2_labs/lab2_C#/ConsoleApp1/StringLibrary.cs
using System;

namespace StringLibrary
{
    public class StringManipulator
    {
        private string value;

        // Конструктор за замовчуванням
        public StringManipulator()
        {
            value = "";
        }

        // Конструктор з параметром
        public StringManipulator(string str)
        {
            value = str;
        }

        // Конструктор копіювання
        public StringManipulator(StringManipulator other)
        {
            value = other.value;
        }

        // Метод отримання довжини рядка
        public int GetLength()
        {
            return value.Length;
        }

        // Метод заміни символу
        public void ReplaceChar(char oldChar, char newChar)
        {
            value = value.Replace(oldChar, newChar);
        }

        // Метод отримання значення рядка
        public string GetValue()
        {
            return value;
        }
    }
}

File: ch2_labs/lab4_C++/ConsoleApp1/StringLibrary.cpp
#include "StringLibrary.h"

// Конструктор за замовчуванням
StringLibrary::StringLibrary() : value("") {}

// Конструктор з параметром
StringLibrary::StringLibrary(const std::string& str) : value(str) {}

// Конструктор копіювання
StringLibrary::StringLibrary(const StringLibrary& other) : value(other.value) {}

// Деструктор
StringLibrary::~StringLibrary() {}

// Отримання довжини рядка
size_t StringLibrary::getLength() const {
    return value.length();
}

// Заміна символу в рядку
void StringLibrary::replaceChar(char oldChar, char newChar) {
    for (char& c : value) {
        if (c == oldChar) c = newChar;
    }
}

// Отримання значення рядка
std::string StringLibrary::getValue() const {
    return value;
}